pkg/operator/operands/prometheus/resources.go

```diff
@@ -69,73 +69,168 @@ func prometheusForKAIConfig(
 	// Check if Prometheus already exists
 	prom, err := common.ObjectForKAIConfig(ctx, runtimeClient, prometheus, mainResourceName, kaiConfig.Spec.Namespace)
 	if err != nil {
-		logger.Error(err, "Failed to check for existing Prometheus instance")
-		return nil, err
+		if !errors.IsNotFound(err) {
+			logger.Error(err, "Failed to check for existing Prometheus instance")
+			return nil, err
+		}
+	}
+	var ok bool
+	prometheus, ok = prom.(*monitoringv1.Prometheus)
+	if !ok {
+		logger.Error(nil, "Failed to cast object to Prometheus type")
+		return nil, fmt.Errorf("failed to cast object to Prometheus type")
 	}
-	prometheus = prom.(*monitoringv1.Prometheus)
 
 	// Set the Prometheus spec from configuration
 	prometheusSpec := monitoringv1.PrometheusSpec{
 		// Basic configuration required for Prometheus Operator to create pods
 		// Using minimal spec to avoid field name issues
 	}
 
-	// Configure TSDB storage if TSDB is configured
-	if config != nil {
-		storageSize, err := config.CalculateStorageSize(ctx, runtimeClient)
-		if err != nil {
-			logger.Error(err, "Failed to calculate storage size")
-			return nil, err
-		}
-		prometheusSpec.Storage = &monitoringv1.StorageSpec{
-			VolumeClaimTemplate: monitoringv1.EmbeddedPersistentVolumeClaim{
-				Spec: v1.PersistentVolumeClaimSpec{
-					StorageClassName: config.StorageClassName,
-					AccessModes:      []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
-					Resources: v1.VolumeResourceRequirements{
-						Requests: v1.ResourceList{
-							v1.ResourceStorage: resource.MustParse(storageSize),
-						},
+	// Configure TSDB storage
+	storageSize, err := config.CalculateStorageSize(ctx, runtimeClient)
+	if err != nil {
+		logger.Error(err, "Failed to calculate storage size")
+		return nil, err
+	}
+	prometheusSpec.Storage = &monitoringv1.StorageSpec{
+		VolumeClaimTemplate: monitoringv1.EmbeddedPersistentVolumeClaim{
+			Spec: v1.PersistentVolumeClaimSpec{
+				StorageClassName: config.StorageClassName,
+				AccessModes:      []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+				Resources: v1.VolumeResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: resource.MustParse(storageSize),
 					},
 				},
 			},
-		}
+		},
+	}
 
-		// Set retention period if specified
-		if config.RetentionPeriod != nil {
-			prometheusSpec.Retention = monitoringv1.Duration(*config.RetentionPeriod)
+	// Set retention period if specified
+	if config.RetentionPeriod != nil {
+		prometheusSpec.Retention = monitoringv1.Duration(*config.RetentionPeriod)
+	}
+
+	// Configure ServiceMonitor selector to match KAI ServiceMonitors
+	if config.ServiceMonitor != nil && *config.ServiceMonitor.Enabled {
+		prometheusSpec.ServiceMonitorSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"accounting": mainResourceName,
+			},
 		}
+		prometheusSpec.ServiceMonitorNamespaceSelector = &metav1.LabelSelector{}
```

Comment:
Maybe use the namespace selector to select all ServiceMonitors in the KAI-scheduler namespace? it will be simpler.